crates/sqlbuiltins/src/functions/table/iceberg/data_files.rs

```diff
@@ -35,8 +35,12 @@ impl ConstBuiltinFunction for IcebergDataFiles {
 
 #[async_trait]
 impl TableFunc for IcebergDataFiles {
-    fn runtime_preference(&self) -> RuntimePreference {
-        RuntimePreference::Remote
+    fn detect_runtime(
+        &self,
+        _args: &[FuncParamValue],
+        _parent: RuntimePreference,
+    ) -> Result<RuntimePreference> {
+        Ok(RuntimePreference::Remote)
```

Comment:
tangent: but shouldn't this technically check the url path to detect the runtime? Same with a few of these other object store related ones. 

The other ones have todo's so it's clear that it's not yet implemented. Can we add that same here? We should also open up an issue to properly detect the runtimes for these functions

Reply:
Yeah, I'll add a TODO here as well.